### DISCUSSION TITLE
If next/link hide the searchParams with `as` props, how to get the value?

### DISCUSSION THREAD

**Discussion by jintak0401** [vercel/next.js#50720] [App Router]
Hi, I am doing migration pages router to app router. During this, I met a problem with `useSearchParams` and `next/link`'s `as` props.

```tsx
<NextLink href={'/blog/1?value=true'} as={'/blog/1'}>
    route button
</NextLink>

// after routing with the NextLink above
const searchParams = useSearchParams();
const value = searchParams.get('value') // it is not true, but null
```

If NextLink hides the href's searchParams with `as` props, I cannot get appropriate value with `useSearchParams().get('value')`. Is there any way to get the value?

**Comment by ilceor** [comment]
Could you solve this? I'm trying to do the same thing
